apps/core/secureaction.cpp

```diff
@@ -0,0 +1,254 @@
+/****************************************************************************
+**
+** Copyright (c) 2018 - 2021 Jolla Ltd.
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "secureaction.h"
+
+#include <QDBusConnection>
+#include <QDBusMessage>
+#include <QDBusObjectPath>
+#include <QDBusPendingCallWatcher>
+#include <QDBusReply>
+#include <QDBusServiceWatcher>
+
+#include <QQmlInfo>
+
+namespace {
+
+const QString deviceLockService = QStringLiteral("org.nemomobile.devicelock");
+const QString authenticatorPath = QStringLiteral("/authenticator");
+const QString authenticatorInterface = QStringLiteral("org.nemomobile.devicelock.Authenticator");
+
+static const QString dbusService = QStringLiteral("org.freedesktop.DBus");
+static const QString dbusPath = QStringLiteral("/org/freedesktop/DBus");
+static const QString dbusInterface = QStringLiteral("org.freedesktop.DBus");
+
+QDBusMessage createAuthenticatorCall(const QString &member, const QVariantList &arguments)
+{
+    QDBusMessage methodCall = QDBusMessage::createMethodCall(
+                deviceLockService,
+                authenticatorPath,
+                authenticatorInterface,
+                member);
+    methodCall.setArguments(arguments);
+    return methodCall;
+}
+
+QString replyPath()
+{
+    static int counter = 0;
+
+    return QStringLiteral("/org/nemomobile/devicelock/client/sailfish_browser/SecureAction%1").arg(++counter);
+}
+
+enum class Method {
+    NoAuthentication    = 0x000,
```

Comment:
OOI is it intentionally 0x000 rather than 0x0000?

Reply:
Not intentional. It will be same in nemo-qml-plugin-devicelock.